test/test.jwt.ts

```diff
@@ -133,6 +133,30 @@ it('can get obtain new access token when scopes are set', (done) => {
   });
 });
 
+
+it('should emit an event for tokens', (done) => {
+  const jwt = new JWT({
+    email: 'foo@serviceaccount.com',
+    keyFile: PEM_PATH,
+    scopes: ['http://bar', 'http://foo'],
+    subject: 'bar@subjectaccount.com'
+  });
+
+  let raisedEvent = false;
+  jwt.on('tokens', tokens => {
+    assert.equal(tokens.access_token, 'initial-access-token');
+    raisedEvent = true;
```

Comment:
do we need this flag? why not just call `done()` here?

Reply:
I feel this test adds some extra assertion on the order of events (`tokens` is expected to be emitted before the 'getAccessToken' callback is called). Even if it's how things are supposed to work, I feel it's wrong to rely on this in the unit test (I can be wrong here of course, don't know the code well enough)

Reply:
You win my best code review feedback of the day medal 🥇 

I simplified and fixed this up quite a bit.